include/engine/guidance/segment_list.hpp

```diff
@@ -235,6 +211,30 @@ void SegmentList<DataFacadeT>::Finalize(const bool extract_alternative,
     if (segments.empty())
         return;
 
+    // check if first two segments can be merged
+    BOOST_ASSERT(segments.size() >= 2);
+    if (segments[0].location == segments[1].location &&
+        segments[1].turn_instruction == extractor::TurnInstruction::NoTurn)
+    {
+        segments[0].travel_mode = segments[1].travel_mode;
+        segments[0].name_id = segments[1].name_id;
+        // Other data??
+        segments.erase(segments.begin() + 1);
+    }
+
+    // announce mode changes
+    for (std::size_t i = 0; i + 1 < segments.size(); ++i)
+    {
+        auto &segment = segments[i];
+        const auto next_mode = segments[i + 1].travel_mode;
+        if (segment.travel_mode != next_mode &&
+            segment.turn_instruction == extractor::TurnInstruction::NoTurn)
+        {
```

Comment:
Not without adding additional information to the edges.

In turning from (a)->(b), we want to know both about the modes of (a) for all information stored along (a), as well as (b). The turn announces the mode change to the mode of (b) at (a). So you either have to store both the current and the next travel mode, or you have to traverse the path to find the travel mode.